### PR TITLE
feature: Add support: yr_scanner_scan_mem_blocks

### PR DESCRIPTION
**Pull request by ikrivosheev**
Feature: https://github.com/Hugal31/yara-rust/issues/42

### CODE REVIEW

**Comment by ikrivosheev** [comment]
@hexgnu, Ready! I added support for: https://yara.readthedocs.io/en/stable/capi.html#c.yr_scanner_scan_mem_blocks . Please, can you review?

File: src/internals/iterator.rs [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+pub struct WrapperMemoryBlockIterator<T> {
+    iter: T,
+    mem_block: std::mem::MaybeUninit<YR_MEMORY_BLOCK>,
+}
+
+impl<T> WrapperMemoryBlockIterator<T> {
+    pub fn new(iter: T) -> Self {
+        Self {
+            iter,
+            mem_block: std::mem::MaybeUninit::uninit(),
```
Comment (ikrivosheev) [inline src/internals/iterator.rs:54]: I am no sure with this. I needed to store current memory_block. After this PR minimal supported rustc version is 1.55

Comment (Hugal31) [reply src/internals/iterator.rs:51]: Why? MaybeUninit is stable since 1.36. Am I missing something?

Comment (ikrivosheev) [reply src/internals/iterator.rs:51]: I use MaybeUninit.write, which was stabilized at 1.55

**Pull request merged**